### DISCUSSION TITLE
Replay workflow instance

### DISCUSSION THREAD

**Discussion by fulganta** [elsa-workflows/elsa-core#2498] [Q&A]
Hi,

I saw that it is planned to do a replay button: https://github.com/elsa-workflows/elsa-core/issues/721
But programmatically speaking can I replay a workflow instance using the workflow id or correlation id?

Thanks

Cyril

**Comment by sfmskywalker** [comment]
Hi @cfrelet ,

Programmatically, you could do the following:

1. Set workflow instance status to `Idle`.
2. Clear its `BlockingActivities` collection.
3. Clear its `Variables` dictionary (taking into account any default variables configured on the workflow definition).
4. Clear its `Input` property (set to `null`).
5. Clear its `ScheduledActivities` collection.
6. Clear its `CurrentActivity` property (set to `null`).
7. Invoke the workflow instance using the `IWorkflowRunner` service.
